Dependency updates, message cleanup, bugfixes, and new features

tadhunt opened the pull request:
* Upgrade Gradle
* Update dependencies to latest versions so it works on Spigot & Paper 1.19.4
* Cleanup and normalize all messages to use MiniMessage formatting
  * Get rid of translateAlternateColorCodes
  * Add more (most?) messages to messages.yml
* Remove sponge support (testing is out of scope for me)
* Add new commands & subcommands so parties & elections can be managed via "/" commands
* Add new feature to open the elections menu when a player hits a block
* bugfix: Table creation happens asynchronously, so SQL errors ('table not found') errors are raised if code tries to query the table and it hasn't yet been created (see issue #125)

Lorenzo0111 commented on build.gradle:
```diff
@@ -25,31 +25,38 @@ import org.apache.tools.ant.filters.ReplaceTokens
     compileTestJava.options.encoding = "UTF-8"
 
-    version = "1.6"
+    version = "1.6.mcplaydates-1.0"
```
Are you able to remove this part?

tadhunt replied:
Yep, will do

tadhunt replied:
fixed in commit 1f2e3cd359e4e1f9070942b4adffb935f1f48b03

Lorenzo0111 commented on elections-api/src/main/java/me/lorenzo0111/elections/api/objects/ElectionBlock.java:
```diff
@@ -1,7 +1,7 @@
  * This file is part of ElectionsPlus, licensed under the MIT License.
  *
- * Copyright (c) Lorenzo0111
+ * Copyright (©) tadhunt
```
Can you fix the copyright?

tadhunt replied:
Sure!

What do you want here?
* Change the © to (c)
* Change tadhunt to Lorenzo0111?
* Both things?

Since I authored this file and there isn't a contributors policy, I wasn't sure what the right thing to do is.

tadhunt replied:
I assumed I should fix both, so those changes are now also in this PR (3dce35260356e0f597e4315daeaf072359131d41)

Lorenzo0111 commented on elections-common/src/main/resources/config.yml:
```diff
@@ -40,3 +40,6 @@ rank:
   strategy: "none" # None, both
+
+update:
+  check: false
```
Please default this to true

tadhunt replied:
will do

tadhunt replied:
fixed in 84eb15f68451a5556cc9098822d5f2a83c28e16b

Lorenzo0111 commented:
Hello and thanks for making a pull request. I love when someone from the community decided to contribute to my projects and this is why I always make my projects opensource.

The code overall looks really good, I’ll test it as soon as I’ll get home. In the meantime are you able to fix the issues I asked in the review above?

tadhunt commented:
> Hello and thanks for making a pull request. I love when someone from the community decided to contribute to my projects and this is why I always make my projects opensource.

Cool!  I needed this for some voting one of our customers wants to do for an building competition they're having, but it didn't work.  I'm more than happy to contribute the updates / new features back for the community.

> The code overall looks really good, I’ll test it as soon as I’ll get home. In the meantime are you able to fix the issues I asked in the review above?

Yep, definitely.  I'm also not sure if some of this (in particular the cmdline arg processing to handle spaces, and how to open the elections gui when a player hits a block) since I'm neither experienced with Java nor Bukkit plugin development.

tadhunt commented:
Hi Lorenzo, following up on whether you plan to merge this work or if you need to see other changes?

Additionally, on my mcplaydates-1.0 branch, I've added several new features:
* Soft Integration with Holographic Displays so to display up-to-date Elections status in holograms
* Soft Integration with GriefPrevention to enable ability to create parties and add them to open elections to vote on when GP claims are transferred to players (I use it for ease of administration of voting on building competitions)
* ability to create "voting blocks" that when hit open the voting menu.

Along the way I found several challenges with the asynchronous nature of the way the cache and database worked, so I've also refactored all of the cache and database logic.  All queries now come directly from the cache, which has been updated with a "write-through" caching model.  The cache is now responsible for persisting dirty entries.

LMK if you'd like any of these changes contributed back, otherwise I'm happy to continue maintaining my fork.

Lorenzo0111 commented:
Hello, I'm sorry for the delay. Are you still interested in this? @tadhunt

tadhunt commented:
Hi @Lorenzo0111, I'd love to be able to roll our changes back into the project so we don't need to maintain a separate fork, but I don't have time in the forseeable future for any development work, so not sure when I could get back to it.

Lorenzo0111 commented:
Closes #125

Lorenzo0111 commented:
#126
